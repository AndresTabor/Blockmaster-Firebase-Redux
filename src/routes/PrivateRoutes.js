import React, { useContext } from 'react'
import { Navigate, Route, Routes } from 'react-router-dom'
import Home from '../components/Home'
import ProfileUser from '../components/ProfileUser'
import { AuthContext } from '../context/authContext'



const PrivateRoutes = ( ) => {
    const { isLoggin } = useContext(AuthContext)
    console.log( isLoggin );
  return (

    <> 
    <Routes>
      <Route path='/' element={<Home/>}/>        
      <Route path='/profile' element={<ProfileUser/>}/> 
      <Route path='/*' element={<Navigate to='/'/>}/>         
    </Routes>
    </>
  )
}

export default PrivateRoutes