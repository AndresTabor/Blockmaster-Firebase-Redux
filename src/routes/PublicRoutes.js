import React from 'react'
import { Route, Routes } from 'react-router-dom'
import Home from '../components/Home';
import Login from '../components/Login';
import Registro from '../components/Registro';




const PublicRoutes = ( ) => {
  // const { isLoggin } = useContext(AuthContext)
  //   console.log( isLoggin );    
  return (
    <>
    <Routes>
      <Route path='/' element={<Home/>}/>
      <Route path='/login' element={<Login/>}/>
      <Route path='/register' element={<Registro/>}/> 
                                                                                 
    </Routes>
    </>
  )
}

export default PublicRoutes