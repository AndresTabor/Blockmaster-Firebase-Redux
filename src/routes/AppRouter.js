import { getAuth, onAuthStateChanged } from 'firebase/auth';
import React, { useEffect, useState } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Login from '../components/Login';
import { AuthContext } from '../context/authContext'
import PrivateRoutes from './PrivateRoutes';
import PublicRoutes from './PublicRoutes';

const AppRouter = () => {
  const [isLoggin, setIsLoggin] = useState()
 // const [checking, setChecking] = useState(true)
  

  useEffect(() => {
    const auth = getAuth();
    onAuthStateChanged(auth, (user) => {
        if(user?.uid){
          console.log(user)
          setIsLoggin(true)
        }
        else{
          console.log('no logueado')
          setIsLoggin(false)
        }
        //setChecking(false)
    })

 
  }, [setIsLoggin])


  // if(checking){
  //   return(
  //       <h1>Espere...</h1>
  //   )
  // }
  
  return <div>
      <AuthContext.Provider value={{ isLoggin }}>
        <BrowserRouter>
        {
          isLoggin === true?
          <PrivateRoutes/> 
          : isLoggin === undefined ?
          console.log("cat")
          :<PublicRoutes/>
        }                                                                                     
        </BrowserRouter>

      </AuthContext.Provider>
  </div>;
};

export default AppRouter;
