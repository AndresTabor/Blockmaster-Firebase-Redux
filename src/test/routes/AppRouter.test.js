import '@testing-library/jest-dom/extend-expect';
import React from "react";
import AppRouter from '../../routes/AppRouter';
import {shallow} from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import PrivateRoutes from '../../routes/PrivateRoutes';
import PublicRoutes from '../../routes/PublicRoutes';
import { BrowserRouter } from 'react-router-dom';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
let store = mockStore();

describe('pruebas <AppRouter />', () => {
    beforeEach(() =>{
        store = mockStore();
    })

    test('validar render de rutas privadas', () => {
        const isLoggin = true;
        const component = shallow(
            <Provider store={store}>
                <AppRouter>                
                    <BrowserRouter>
                    {
                    isLoggin === true?
                    <PrivateRoutes/> 
                    : isLoggin === undefined ?
                    console.log("cat")
                    :<PublicRoutes/>
                    }                                                                                     
                    </BrowserRouter>
                
                </AppRouter>
            </Provider>
        )   
        expect(component).toMatchSnapshot()
        expect(component.find('PrivateRoutes').length).toBe(1)
    })
    test('validar render de rutas publicas', () => {
        const isLoggin = false;
        const component = shallow(
            <Provider store={store}>
                <AppRouter>
                
                    <BrowserRouter>
                    {
                    isLoggin === true?
                    <PrivateRoutes/> 
                    : isLoggin === undefined ?
                    console.log("cat")
                    :<PublicRoutes/>
                    }                                                                                     
                    </BrowserRouter>
                
                </AppRouter>
            </Provider>
        )           
        expect(component).toMatchSnapshot();
        expect(component.find('PublicRoutes').length).toBe(1)        
    })
})