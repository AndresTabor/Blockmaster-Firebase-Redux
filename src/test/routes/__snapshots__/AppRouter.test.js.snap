// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas <AppRouter /> validar render de rutas privadas 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <AppRouter>
    <BrowserRouter>
      <PrivateRoutes />
    </BrowserRouter>
  </AppRouter>
</ContextProvider>
`;

exports[`pruebas <AppRouter /> validar render de rutas publicas 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <AppRouter>
    <BrowserRouter>
      <PublicRoutes />
    </BrowserRouter>
  </AppRouter>
</ContextProvider>
`;
